### PR TITLE
Add ts typings

### LINKED ISSUE #12: Typescript definitions
Hello!

First of all thanks for this awesome tool! I needed exactly this tool :)

Having said that, I was wondering if your open on hosting typescript definitions inside the repository. I can get them done and push a PR if you are fine with it.

Thanks!

### PR DESCRIPTION
**Pull request by bilby91**
Closes #12

### CODE REVIEW

**Comment by bilby91** [comment]
@paazmaya I exposed the methods in the README, those seem to be the public API. If you think we are missing something let me know please!

File: package.json [hunk 1]
```diff
@@ -23,9 +25,11 @@
     "node": ">=6.9.5"
   },
   "dependencies": {
+    "@types/node": "^9.4.6",
     "check-type": "^0.4.11",
     "js-yaml": "^3.10.0",
-    "optionator": "^0.8.2"
+    "optionator": "^0.8.2",
+    "typescript": "^2.7.2"
```
Comment (paazmaya) [inline package.json]: Are these needed always? Perhaps just in the development environment?

Comment (bilby91) [reply package.json]: I probably used the wrong command. They should be devDependencies. I'll fix it.

**Comment by bilby91** [comment]
@paazmaya sorry for the delay. Moved dependencies to devDependencies.

**Comment by paazmaya** [comment]
Thank you for the contribution! Here is some 🍇 for you.

**Pull request merged**